### PR TITLE
Implemented Dog class

### PR DESCRIPTION
**Pull request by caoalbe**
created a dog class with an age (1 human year is 8 dog years)!

### CODE REVIEW

File: src/main/java/Dog.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+public class Dog implements Domesticatable, Tradable {
+    private int age;
+
+    public Dog(int age) {
+        this.age = age+1;
```
Comment (marianmwang) [inline src/main/java/Dog.java]: why is the dog aging

File: src/main/java/Dog.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public Dog() {
+        this.age = 0;
+    }
+
+    public void birthday() {
+        // because 1 human year is 8 dog years :)
```
Comment (marianmwang) [inline src/main/java/Dog.java:13]: upvoted for good comment

File: src/main/java/Dog.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @Override
+    public int getPrice() {
+        return 5;
+    }
+
+    @Override
+    public String toString() {
+        return "A cute dog that is " + (this.age - 1) + " dog years old!";
```
Comment (marianmwang) [inline src/main/java/Dog.java]: why is it -1? also i think you are actually giving the dog's age in human years, not dog years

File: src/test/java/DogTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import org.junit.*;
+
+import static org.junit.Assert.*;
+
+public class DogTest {
+    Dog d;
+
+    @Before
+    public void setUp() throws Exception {
+        d = new Dog();
```
Comment (marianmwang) [inline src/test/java/DogTest.java:11]: you should also test Dog(int) since you have another constructor for it

**Review by marianmwang — CHANGES_REQUESTED** [review]
error in dog age, other minor errors

File: src/main/java/Dog.java (hunk 2 above)
Comment (caoalbe) [reply src/main/java/Dog.java:13]: :)

File: src/main/java/Dog.java (hunk 1 above)
Comment (Silk8Worm) [reply src/main/java/Dog.java]: Poor style. Storing the age as +1, then printing -1. Instead, store the age as specified.

File: src/test/java/DogTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @Test(timeout = 50)
+    public void TestGetPrice() {
+        assertEquals(5, d.getPrice());
+    }
+
+    @Test(timeout = 50)
+    public void TestGetAge() {
+        assertEquals(0, d.getAge());
```
Comment (Silk8Worm) [inline src/test/java/DogTest.java:28]: this test would fail with the original this.age = age+1

**Pull request merged**